src/php/Compiler/Domain/Emitter/StatementEmitter.php

```diff
@@ -25,33 +25,37 @@ public function emitNode(AbstractNode $node, bool $enableSourceMaps): EmitterRes
         $this->outputEmitter->resetIndentLevel();
         $this->outputEmitter->resetSourceMapState();
 
-        $sourceLocation = $node->getStartSourceLocation();
-        $file = $sourceLocation
-            ? $sourceLocation->getFile()
-            : 'string';
+        return new EmitterResult(
+            $enableSourceMaps,
+            $this->phpCode($node),
+            $this->sourceMap($enableSourceMaps),
+            $this->source($node),
+        );
+    }
 
+    private function phpCode(AbstractNode $node): string
+    {
         ob_start();
         $this->outputEmitter->emitNode($node);
-        $code = ob_get_clean();
 
+        return ob_get_clean();
```

Comment:
@Chemaclass do you think could make sense to move these PHP functions (`ob_start()` & `ob_get_clean()`) to an interface?

Reply:
That's a really interesting idea for a follow up PR 💡